source/smart.c

```diff
@@ -123,17 +123,71 @@ int getText(unsigned char *T, char *path, int FREQ[SIGMA], int TSIZE) {
 	}
 	else printf("\tError in loading text buffer. No index file exists.\n");
 	T[i]='\0';
-	// compute the frequency of characters and the dimension of the alphabet
-	int nalpha = 0;
-	int maxcode= 0;
-	for(j=0; j<SIGMA; j++) FREQ[j]=0;
-	for(j=0; j<i; j++) {
-		if(FREQ[T[j]]==0) nalpha++;
-		FREQ[T[j]]++;
-		if(maxcode<T[j]) maxcode=T[j];
-	}
-	printf("\tAlphabet of %d characters.\n", nalpha);
-	printf("\tGreater chararacter has code %d.\n", maxcode);
+    // compute the frequency of characters and the dimension of the alphabet
+    int nalpha = 0;
+    int nbigram = 0;
+    int ntrigram = 0;
+    int maxcode= 0;
+    for(j=0; j<SIGMA; j++) FREQ[j]=0;

```

Comment:
would be better to wrap this into a function like
```C++
void initArr(int *arr, int length) {
  for(int i = 0; i < length; i++) {
    arr[i] = 0;
  }
}
```

and reuse it any time you have to initialize an arr? an inline for can be confusing sometimes